caffe/include/caffe/layers/molgrid_data_layer.hpp

```diff
@@ -310,19 +325,107 @@ class BaseMolGridDataLayer : public MolGridDataLayer<Dtype> {
   void getLigandGradient(int batch_idx, vector<float3>& gradient);
   void getMappedLigandGradient(int batch_idx, unordered_map<string, float3>& gradient);
 
+  std::vector<std::string> getRecTypes() {
```

Comment:
Can we move these to the cpp file?

Reply:
Also, use the molgrid names - you can get a vector of type names from the atom typer